pkg/logs/agent.go

```diff
@@ -61,22 +63,47 @@ func NewAgent(sources *config.LogSources, services *service.Services, processing
 	// setup the pipeline provider that provides pairs of processor and sender
 	pipelineProvider := pipeline.NewProvider(config.NumberOfPipelines, auditor, diagnosticMessageReceiver, processingRules, endpoints, destinationsCtx)
 
+	containerLaunchables := []container.Launchable{
+		{
+			IsAvailable: docker.IsAvailable,
+			Launcher: func() restart.Restartable {
+				return docker.NewLauncher(
+					time.Duration(coreConfig.Datadog.GetInt("logs_config.docker_client_read_timeout"))*time.Second,
+					sources,
+					services,
+					pipelineProvider,
+					auditor,
+					coreConfig.Datadog.GetBool("logs_config.docker_container_use_file"),
+					coreConfig.Datadog.GetBool("logs_config.docker_container_force_use_file"))
+			},
+		},
+		{
+			IsAvailable: kubernetes.IsAvailable,
+			Launcher: func() restart.Restartable {
+				return kubernetes.NewLauncher(sources, services, coreConfig.Datadog.GetBool("logs_config.container_collect_all"))
+			},
+		},
+	}
+
+	// when k8s_container_use_file is true, always attempt to use the kubernetes launcher first
+	if coreConfig.Datadog.GetBool("logs_config.k8s_container_use_file") {
+		containerLaunchables[0], containerLaunchables[1] = containerLaunchables[1], containerLaunchables[0]
+	}
+
 	// setup the inputs
 	inputs := []restart.Restartable{
 		file.NewScanner(sources, coreConfig.Datadog.GetInt("logs_config.open_files_limit"), pipelineProvider, auditor, file.DefaultSleepDuration),
-		container.NewLauncher(
-			coreConfig.Datadog.GetBool("logs_config.container_collect_all"),
-			coreConfig.Datadog.GetBool("logs_config.k8s_container_use_file"),
-			coreConfig.Datadog.GetBool("logs_config.docker_container_use_file"),
-			coreConfig.Datadog.GetBool("logs_config.docker_container_force_use_file"),
-			time.Duration(coreConfig.Datadog.GetInt("logs_config.docker_client_read_timeout"))*time.Second,
-			sources, services, pipelineProvider, auditor),
 		listener.NewLauncher(sources, coreConfig.Datadog.GetInt("logs_config.frame_size"), pipelineProvider),
 		journald.NewLauncher(sources, pipelineProvider, auditor),
 		windowsevent.NewLauncher(sources, pipelineProvider),
 		traps.NewLauncher(sources, pipelineProvider),
 	}
 
+	// Only try to start the container launchers if we are in a container environment or docker is available
+	if coreConfig.IsContainerized() || coreConfig.IsFeaturePresent(coreConfig.Docker) || coreConfig.IsFeaturePresent(coreConfig.Kubernetes) {
```

Comment:
I think using `coreConfig.IsContainerized()` is not necessary. It would prevent on-host install with Docker to trigger.